### PR TITLE
LG-3241, LG-3194: controlled toast and notification bar updates

### PR DESCRIPTION
**Pull request by shaneeza**
## ✍️ Proposed changes

[Demo](https://shaneeza.github.io/leafygreen-ui/?path=/story/components-toast--with-initial-toasts)

🎟 _Jira ticket:_ [LG-3241](https://jira.mongodb.org/browse/LG-3241)
🎟 _Jira ticket:_ [LG-3194](https://jira.mongodb.org/browse/LG-3194)

### For bug fixes, new features & breaking changes

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

### CODE REVIEW

File: packages/toast/src/ControlledToast/ControlledToast.spec.tsx [hunk 1]
```diff
@@ -70,6 +70,24 @@ describe('packages/toast/controlled', () => {
       expect(toast).toBeInTheDocument();
     });
 
+    test('does not render when `open` is true and component is unmounted', async () => {
```
Comment (TheSonOfThomp) [inline packages/toast/src/ControlledToast/ControlledToast.spec.tsx:73]: We should also test that it un-renders when the render condition toggles

**Review by TheSonOfThomp — COMMENTED** [review]
Just needs the extra test condition

File: packages/toast/src/ControlledToast/ControlledToast.spec.tsx [hunk 2]
```diff
@@ -70,6 +70,58 @@ describe('packages/toast/controlled', () => {
...
+            />
+          )}
+        </ToastProvider>,
+      );
+
+      const toast = await waitFor(() => queryByTestId('test-toast1b'));
+      expect(toast).not.toBeInTheDocument();
+    });
+
+    test('unmounts when `open` is true and component render is toggled to false', async () => {
```
Comment (shaneeza) [inline packages/toast/src/ControlledToast/ControlledToast.spec.tsx:91]: @TheSonOfThomp I'm not sure if this is the best way to test this. To get it to work i had to change the data-testid for the rerendered Toast so I'm not sure if that is correct. Do you have any suggestions?

File: packages/toast/src/InternalToast/InternalToast.spec.tsx [hunk 3]
```diff
@@ -195,8 +195,9 @@ describe('packages/toast/internal-toast', () => {
 });
 
 // eslint-disable-next-line jest/no-disabled-tests
-describe.skip('TS types', () => {
-  render(
+describe.skip('TS types behave as expected', () => {
```
Comment (shaneeza) [inline packages/toast/src/InternalToast/InternalToast.spec.tsx:198]: Updated this to eliminate some of the warnings in the test console.

File: packages/toast/src/ControlledToast/ControlledToast.spec.tsx (hunk 2 above)
Comment (TheSonOfThomp) [reply packages/toast/src/ControlledToast/ControlledToast.spec.tsx:91]: I don't think in this case we're checking for the same component. I think after the `rerender` we want to 
```
await waitForElementToBeRemoved(toast)
expect(toast).not.toBeInTheDocument()
```

File: packages/toast/src/ControlledToast/ControlledToast.spec.tsx [hunk 4]
```diff
@@ -88,6 +88,40 @@ describe('packages/toast/controlled', () => {
       expect(toast).not.toBeInTheDocument();
     });
 
+    test('unmounts when `open` is true and component render is toggled to false', async () => {
+      let isOpen = true;
```
Comment (TheSonOfThomp) [inline packages/toast/src/ControlledToast/ControlledToast.spec.tsx:92]: nit: I'd call this var `shouldRender`, to avoid confusion

**Review by TheSonOfThomp — APPROVED** [review]

**Pull request merged**